test/dns_dnscmd/dnscmd_test.rb

```diff
@@ -13,63 +31,98 @@ def test_dnscmd_provider_initialization
   end
 
   def test_overwrite_ptr_record
-    Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('192.168.33.33').returns('host.domain')
-    Proxy::Dns::Dnscmd::Record.new.create_ptr_record('host.domain', '192.168.33.33')
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('192.168.33.33').returns('host.foo.bar.domain.local')
+    @server.create_ptr_record('host.foo.bar.domain.local', '192.168.33.33')
   end
 
   def test_create_duplicate_ptr_record_fails
-    Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('192.168.33.33').returns('another.host.domain')
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('192.168.33.33').returns('another.host.foo.bar.domain.local')
     assert_raise Proxy::Dns::Collision do
-      Proxy::Dns::Dnscmd::Record.new.create_ptr_record('host.domain', '192.168.33.33')
+      @server.create_ptr_record('host.foo.bar.domain.local', '192.168.33.33')
     end
   end
 
   def test_remove_address_record
-    Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('host.domain').returns(true)
-    Proxy::Dns::Dnscmd::Record.any_instance.expects(:execute).with('/RecordDelete domain host.domain. A /f', anything).returns(true)
-    assert Proxy::Dns::Dnscmd::Record.new.remove_a_record('host.domain')
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('host.foo.bar.domain.local').returns(true)
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:execute).with('/RecordDelete bar.domain.local host.foo.bar.domain.local. A /f', anything).returns(true)
+    assert @server.remove_a_record('host.foo.bar.domain.local')
   end
 
   def test_remove_non_existent_address_record_raises_exception
-    Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('host.domain').returns(false)
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('host.foo.bar.domain.local').returns(false)
     assert_raise Proxy::Dns::NotFound do
-      Proxy::Dns::Dnscmd::Record.new.remove_a_record('host.domain')
+      @server.remove_a_record('host.foo.bar.domain.local')
     end
   end
 
   def test_remove_ptr_record
     Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('192.168.33.33').returns(true)
-    assert Proxy::Dns::Dnscmd::Record.new.remove_ptr_record('192.168.33.33')
+    assert @server.remove_ptr_record('192.168.33.33')
   end
 
   def test_remove_nonexistent_ptr_record_raises_exception
     Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('192.168.33.33').returns(false)
     assert_raise Proxy::Dns::NotFound do
-      Proxy::Dns::Dnscmd::Record.new.remove_ptr_record('192.168.33.33')
+      @server.remove_ptr_record('192.168.33.33')
     end
   end
+
+  def test_dns_zone_matching_edge_cases
+    assert_equal('33.168.192.in-addr.arpa', @server.match_zone('33.33.168.192.in-addr.arpa'))
+    assert_equal('domain.local', @server.match_zone('bar.domain.local'))
```

Comment:
`bar.domain.local` is the host name in this case. As `bar.domain.local` is already a zone, I am guessing the user's intention  is to rather add it to the next best match (we talked about it, dnscmd adds a `(same as parent)` A record in this case).

The alternative would be to raise `Proxy::Dns::Collision`. But it is still an edge case when we have  both zones available; I'd rather not fail. What do you think?